straxen/plugins/peaklet_processing.py

```diff
@@ -266,13 +267,14 @@ def peak_saturation_correction(records, peaks, to_pe,
 
     # Buff the sum wf [pe] of non-saturated channels
     b_sumwf = np.zeros(len_buffer, dtype=np.float32)
-    # Buff the records data [ADC] in saturated channels
-    b_pulse = np.zeros((n_channels, len_buffer), dtype=np.int16)
+    # Buff the records 'data' [ADC] in saturated channels
+    b_pulse = np.zeros((n_channels, max_nrecord), dtype=np.int16)
```

Comment:
@jorana I am not sure what do you mean by changing from len_buffer to max_nrecords. The b_pulse is supposed to save the whole pulse with 10ns resolution.

Reply:
Sorry Tianyo, my bad, should have been the line below: https://github.com/XENONnT/straxen/pull/244/commits/9863e2b6ddf9a525b074ebcf3fc81c7f93593876